File: app/src/main/java/im/point/dotty/db/PostDao.java
package im.point.dotty.db;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;

import java.util.List;

import im.point.dotty.model.Post;
import io.reactivex.Observable;

@Dao
public interface PostDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void addPosts(Post... posts);

    @Query("SELECT * FROM posts WHERE user_id = :userId ORDER BY timestamp")
    Observable<List<Post>> getUserPosts(long userId);

    @Query("SELECT * FROM posts WHERE id = :id")
    Observable<Post> getPost(long id);
}

File: app/src/main/java/im/point/dotty/db/PostDao.kt
/*
 * Copyright (c) 2019-2021 Pavel A. Sokolov
 */
package im.point.dotty.db

import androidx.room.Dao
import androidx.room.Query
import im.point.dotty.model.Post
import kotlinx.coroutines.flow.Flow

@Dao
interface PostDao : CommonDao<Post> {
    @Query("SELECT * FROM posts WHERE id = :id")
    fun getItem(id: String): Post?

    @Query("SELECT * FROM posts WHERE id = :id")
    fun getItemFlow(id: String): Flow<Post?>

    @Query("SELECT * FROM posts")
    fun getAllFlow(): Flow<List<Post>>
}